etna/analysis/forecast/utils.py

```diff
@@ -24,17 +25,13 @@ def get_residuals(forecast_df: pd.DataFrame, ts: "TSDataset") -> "TSDataset":
 
     Returns
     -------
-    new_ts: TSDataset
+    new_ts:
         TSDataset with residuals in forecasts
 
     Raises
```

Comment:
I don't really understand how should this function work if there are target components and hierarchy.

I also found that `make_future` doesn't handle target components and hierarchy.

Reply:
You can document constraints here